### PR TITLE
AMP support for Numpy ops

### PR DESCRIPTION
**Pull request by mk-61**
## Description ##
1. Add all registered ops (mostly Numpy) to AMP lists
2. Removed "skipped" attribute from AMP tests
3. Some change in AMP monkey patching, since NumPy ops require a bit different handling.

### CODE REVIEW

**Comment by mk-61** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu]

**Comment by mk-61** [comment]
@mxnet-bot run ci [all]

**Comment by szha** [comment]
```
[2020-09-24T01:42:24.553Z] ************* Module mxnet.contrib.amp.loss_scaler

[2020-09-24T01:42:24.553Z] python/mxnet/contrib/amp/loss_scaler.py:45:4: C0111: Missing method docstring (missing-docstring)

[2020-09-24T01:42:24.553Z] python/mxnet/contrib/amp/loss_scaler.py:52:8: W0105: String statement has no effect (pointless-string-statement)
```

**Comment by mk-61** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by mk-61** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu, unix-gpu]

**Comment by mk-61** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu]

**Comment by mk-61** [comment]
@mxnet-bot run ci [windows-gpu, macosx-x86_64]

**Review by sxjscience — APPROVED** [review]
I took a look at the PR and it looks good.

File: tests/python/gpu/test_contrib_amp.py [hunk 1]
```diff
@@ -66,45 +66,39 @@ def test_amp_coverage(amp_tests):
...
+    - If you are not sure which list to choose, FP32_FUNCS is the
+                     safest option"""
+    diff = required - covered
+    assert not diff, f"{len(diff)} operators {sorted(diff)} do not exist in AMP lists (in " \
+        f"python/mxnet/contrib/amp/lists/symbol_fp16.py) - please add them. " \
+        f"\n{guidelines}"
 
 @with_seed()
-@pytest.mark.skip(reason='Error during waitall(). Tracked in #18099')
+@pytest.mark.skip(reason="WIP")
```
Comment (ptrendx) [inline tests/python/gpu/test_contrib_amp.py:101]: ?

**Review by ptrendx — APPROVED** [review]
LGTM, modulo this change in the reason for the skipped test - could you clarify that?

**Pull request merged**